### ISSUE TITLE
Disabling HTML5 validation as suggested in FAQ causes problem with showing error messages

### ISSUE THREAD

**Issue by klis87** [davidkpiano/react-redux-form#889]
### The Problem
references #823 

I have following components:
```javascript
import React from 'react';
import { Control } from 'react-redux-form';

import FormGroup from './FormGroup';

const isValid = ({ fieldValue }) => fieldValue.valid || (!fieldValue.touched && !fieldValue.asyncKeys);

const Input = props => (
  <Control
    {...props}
    component={FormGroup}
    mapProps={{ valid: isValid }}
  />
);

export default Input;
```
```javascript
import React from 'react';
import { FormControl, FormGroup as BootstrapFormGroup, ControlLabel } from 'react-bootstrap';
import { Errors } from 'react-redux-form';

import ErrorMessages from './ErrorMessages';

const show = field => field.touched || field.asyncKeys;
const onInvalid = e => e.preventDefault();

const FormGroup = ({ valid, label, messages, required, ...rest }) => (
  <BootstrapFormGroup validationState={valid ? null : 'error'}>
    <ControlLabel>{label}</ControlLabel>
    <FormControl {...rest} required={required} onInvalid={onInvalid} />
    <Errors
      model={rest.name}
      messages={messages}
      component={ErrorMessages}
      show={show}
    />
  </BootstrapFormGroup>
);

export default FormGroup;
```
I want to disable HTML5 validation. Previously I used form `novalidate`, but it was recently change to match HTML spec. Now, recommended approach is to use `onInvalid = e => e.preventDefault()`.

The problem is, that I wanted error state and error messages to be visible only when field is touched. It won't work in the following case - you don't fill your form but just press submit immediately - `show` and `isValid` aren't called when clicking submit, so my components aren't told that fields should become touched and no message or invalid styles are being shown.

What is important here, to be able to reproduce it, when your form is blank and you click submit, you need to have at least 1 field violating HTML5 validation, the easiest of cource to play with is field with `required` prop.

This is really a serious issue as currently there is no way to make this work, and all I can do is to revert to older version.  

### Steps to Reproduce
Submiting a form when at least 1 field doesn't pass HTML validation like `required`.

### Expected Behavior
`show` and `isValid` should be called `onSubmit` event

### Actual Behavior
`show` and `isValid` are not called `onSubmit` event

**Comment by peterox** [comment]
Hi @klis87,

I totally agree that this change was not a good one. For a library that provides validation options over and above the HTML5 validations, to then behave the same way as the HTML5 spec seems counter productive.

FYI to get around this problem, remove the 'onInvalid' attributes as stated in the FAQ and instead add the 'formNoValidate' attribute to your submit buttons (NOT your form).

As long as react-redux-form doesn't start complying with the HTML5 spec in this regard then HTML5 validations will not be triggered BUT your validations will be triggered when the user clicks the submit button.

Peter

**Comment by davidkpiano** [comment]
@klis87 @peterox I think I have a solution.

What if there were a new prop on `<Form>`, called `hideNativeErrors`? The behavior of this would be:

- No native HTML5 error bubbles will be shown when any control is invalid.
- The form will **not** submit if a control is invalid (this is different behavior than `noValidate`, which will always submit a form)
- Custom error messages (using `<Errors>`) will show as normal.

I think this was a suggestion of @peterox. What do you think?

Two 👍 's and I'll get this in ASAP.

**Comment by klis87** [comment]
Thanks, that was really quick!
Could you please upload this new version to npm?

**Comment by davidkpiano** [comment]
I will publish this by end of day today.

**Comment by davidkpiano** [comment]
Published as 1.14.0!

**Comment by klis87** [comment]
Thank you! And i confirm, everything works properly with the new version :+1:
